.github/workflows/prod-cf-deploy-lambda.yml

```diff
@@ -17,7 +17,11 @@ defaults:
 
 jobs:
   deploy-lambda:
-    if: github.ref == 'refs/heads/main'
+    # We need to restrict this workflow to only run on releases on the main
```

Comment:
Can you help me understand why we don't need a branch check on the workflow? Would the lack of a check there mean someone could workflow_dispatch any branch into our stage environment and not just main?

This change you are introducing looks correct to me fwiw, but when comparing to staging I realized I either may not understand what this is doing or we may want to further protect staging from having feature branches deployed?

Reply:
Let's see if this makes sense (to me and also to you): This workflow can be triggered by either (a) a release or (b) workflow_dispatch.

For (a) (a release), the GitHub Context will include the `github.event.release.target_commitish` field and its value will be the name of branch on which the release was tagged. The conditional statement at the start of the `deploy-lambda` job will check to see if the branch name is `main` and only run this workflow if that is true. So, a release tagged on any other branch will get skipped by this conditional test. In the case of a release trigger, the `github.ref` context is a reference to the tag (e.g., `refs/tags/<tag_name>`.

For (b) (a workflow_dispatch trigger), a user in GitHub can attempt to run this from any branch via the Actions web UI. The first thing that the `deploy-lambda` job does is check to see if it's being run on the `main` branch by checking the `github.ref` field. So, if the manual trigger is done from a feature branch, the job will be skipped because the `github.ref` value isn't `main`.

Does this make sense and does it address your concern?

The [stage-cf-deploy-lambda.yml)[https://github.com/MITLibraries/cf-lambda-custom-domain/blob/main/.github/workflows/stage-cf-deploy-lambda.yml] has both `on.workflow_dispatch` and `on.push` triggers. In that workflow, we don't do any branch checks for the `on.workflow_dispatch` trigger which **does** mean that a user can manually deploy any feature branch into the Stage-Workloads environment. But maybe we don't want that?

Reply:
Your last statement is what I was asking about. _Should_ we allow any branch to deploy to staging? That feels incorrect, but is ultimately an infrastructure question I'll let you work out with your team. Personally, I feel like the staging workflow should have `if: github.ref == 'refs/heads/main'` added to protect the staging environment from having arbitrary branches deployed from workflow dispatch. However, it's possible this is an intentional ability to deploy from feature branches prior to merge.

Regardless, we both agree on what this is doing and it feels correct. I was just exploring further to understand how this differed from other environments and was confused as to how stage was being protected if it didn't have this type of check (and your answer seems to be "it is not").

Reply:
I agree with you that this is a good question about what can/should be deployable into Stage-Workloads. Considering your questions, I'm leaning toward more tightly restricting the `workflow_dispatch` run ability for the stage deploy. If we run into roadblocks in the future where that actually seems to be bad, we can loosen things up later. 

I know that you just approved this PR, but I'm going to take the time to update the stage workflow to address this now and ask you for one more review today or tomorrow (pushing the additional commit will cancel out your current approval).